src/components/Pagination/index.js

```diff
@@ -12,46 +15,98 @@ import {
-import { StyledNavLink } from "./styled";
-import { useLocation, useParams } from "react-router-dom/cjs/react-router-dom.min";
+import { selectMoviesTotalPages } from "../../features/movies/moviesSlice";
+import { selectPeopleTotalPages } from "../../features/people/peopleSlice";
 
-export const Pagination = () => {
+export const Pagination = ({ noDisplay }) => {
+  const { page } = useParams();
   const firstMoviePage = useSelector(selectFirstMoviePage)
   const lastMoviePage = useSelector(selectLastMoviePage);
-  const { page } = useParams();
-
-  let pageNumber = +page;
-  const previousPage = pageNumber <= firstMoviePage ? pageNumber : pageNumber - 1;
-  const nextPage = pageNumber >= lastMoviePage ? pageNumber : pageNumber + 1;
-  const isFirstPage = pageNumber === firstMoviePage ? true : false;
-  const isLastPage = pageNumber === lastMoviePage ? true : false;
+  const firstPeoplePage = useSelector(selectFirstPeoplePage)
+  const lastPeoplePage = useSelector(selectLastPeoplePage);
+  const firstSearchPage = useSelector(selectFirstSearchPage)
+  const lastSearchPage = useSelector(selectLastSearchPage);
   const location = useLocation();
   const path = location.pathname.split("/")[1];
+  let firstPage;
+  let lastPage;
+  let pageNumber = +page;
+  const moviesTotalPages = useSelector(selectMoviesTotalPages);
+  const peopleTotalPages = useSelector(selectPeopleTotalPages);
+  const moviesQueryToDisplay = useSelector(selectMoviesQueryToDisplay)
+  const PeopleQueryToDisplay = useSelector(selectPeopleQueryToDisplay)
+
+  console.log("moviesQueryToDisplay", moviesQueryToDisplay)
+  console.log("PeopleQueryToDisplay", PeopleQueryToDisplay)
+  console.log("moviesTotalPages", moviesTotalPages)
+  console.log("peopleTotalPages", peopleTotalPages)
+
+  if (moviesQueryToDisplay || PeopleQueryToDisplay) {
+    firstPage = firstSearchPage;
+    if (path === "movies") {
+      lastPage = moviesTotalPages ? moviesTotalPages : lastSearchPage;
+    } else if (path === "people") {
+      lastPage = peopleTotalPages ? peopleTotalPages : lastSearchPage;
+    }
+  } else {
+    if (path === "movies") {
+      firstPage = firstMoviePage;
+      lastPage = lastMoviePage
+    };
+    if (path === "people") {
+      firstPage = firstPeoplePage;
+      lastPage = lastPeoplePage
+    }
+  }
+
+  const previousPage = pageNumber <= firstPage ? pageNumber : pageNumber - 1;
+  const nextPage = pageNumber >= lastPage ? pageNumber : pageNumber + 1;
+  const isFirstPage = pageNumber === firstPage ? true : false;
+  const isLastPage = pageNumber === lastPage ? true : false;
+  const moviesQuery = useSelector(selectMoviesQuery);
+  const peopleQuery = useSelector(selectPeopleQuery);
+
+  let queryPath;
+  if (moviesQuery && !peopleQuery) {
+    queryPath = `?${searchInputParamName}=${moviesQuery}`;
+  } else if (!moviesQuery && peopleQuery) {
+    queryPath = `?${searchInputParamName}=${peopleQuery}`;
+  } else {
+    queryPath = "";
+  }
```

Comment:
This part is a solution that I needed, but I couldn't figure it out. Damn, great job!

Reply:
Right, this here is impressive